File: RetroGameGauntlet.Forms/ViewModels/SearchPlatformsViewModel.cs
using System.Collections.Generic;
using System.Linq;
using RetroGameGauntlet.Forms.Services;
using Xamarin.Forms;

namespace RetroGameGauntlet.Forms.ViewModels
{
    public class SearchPlatformsViewModel : BaseViewModel
    {
        private string _searchText;
        public string SearchText
        {
            get { return _searchText; }
            set
            {
                _searchText = value;
                RaisePropertyChanged(nameof(SearchText));
                Handle_SearchTextChanged();
            }
        }

        public bool HasItems
        {
            get { return Games?.Any() == true; }
        }

        private List<KeyValuePair<string, string>> _games;
        public List<KeyValuePair<string, string>> Games 
        { 
            get { return _games; }
            set
            {
                _games = value;
                RaisePropertyChanged(nameof(Games), nameof(HasItems));
            }
        }

        private readonly IPlatformLoaderService _platformLoader;

        public SearchPlatformsViewModel()
        {
            _platformLoader = DependencyService.Get<IPlatformLoaderService>();
        }

        private void Handle_SearchTextChanged()
        {
            Games = string.IsNullOrEmpty(SearchText)
                          ? null
                          : _platformLoader.FindGames(SearchText);
        }

   }
}

File: RetroGameGauntlet.PCL/ViewModels/SearchPlatformsViewModel.fs
namespace RetroGameGauntlet.PCL.ViewModels

open System
open System.Collections.Generic
open RetroGameGauntlet.PCL.Services
open RetroGameGauntlet.PCL.ViewModels.Items

/// The "Search" page view model.
type SearchPlatformsViewModel(platformLoader: IPlatformLoaderService) =
    inherit ViewModelBase()
    let mutable games: seq<GameItemViewModel> = Seq.empty
    let mutable searchText: string = String.Empty

    /// The listview source.
    member this.Games
        with get() = games
        and set parameter =
            games <- parameter
            this.NotifyPropertyChanged <@ this.Games @>
            this.NotifyPropertyChanged <@ this.HasItems @>

    /// A member describes if there is any game in the listview source.
    member this.HasItems = not (this.Games |> Seq.isEmpty)

    /// The serach filter.
    member this.SearchText
        with get() = searchText
        and set parameter =
            searchText <- parameter
            this.NotifyPropertyChanged <@ this.SearchText @>
            if String.IsNullOrEmpty searchText then
                this.Games <- Seq.empty
            else 
                async {
                    let! newGames = platformLoader.FindGamesForAsync(searchText)
                    this.Games <- newGames |> Seq.map(fun x -> GameItemViewModel(x))
                }
                |> Async.StartImmediate